c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1549,8 +1559,9 @@ class DeviceCachingAllocator {
 
       // splitting a block depends on `max_split_size`, which may have changed
       // between whe checkpoint was taken and now, so we make sure to recreate
-      // the behavior from the checkpoint.
-      bool split = (i + 1) < segment.blocks.size();
+      // the behavior from the checkpoint. Keep splitting as long as there is
```

Comment:
nit: rephrase to add "why" we keep splitting: since we are reconstructing the state of the allocator, blocks are already rounded to how they would appear inside the segment, so any leftover space must be split into another block.